src/main/java/seedu/address/ui/TaskTabPanel.java

```diff
@@ -140,11 +140,11 @@ public void switchTabOnClick() {
             CommandResult commandResult;
 
             Tab currTaskTab = getTaskTabPane().getSelectionModel().getSelectedItem();
-            if (currTaskTab.equals(allTasksTab)) {
+            if (currTaskTab.idProperty() == allTasksTab.idProperty()) {
```

Comment:
no don't use reference comparison pls. Compare `getId` pls if you want compare `id`